src/osemgrep/language_server/Unit_LS.ml

```diff
@@ -91,6 +92,17 @@ let add_file ?(git = false) ?(dirty = false)
   if (not dirty) && git then Git_wrapper.commit workspace "test";
   file
 
+let with_mock_envvars f () =
+  let old_settings = !Semgrep_envvars.v in
+  let new_settings = { old_settings with app_token = Some "123456789" } in
+  Semgrep_envvars.v := new_settings;
```

Comment:
You can use Common.save_excursion (which is also exn-friendly)

Reply:
Forgot about this again, will change